packages/angular/ssr/src/render.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {
+  ApplicationRef,
+  StaticProvider,
+  Type,
+  ɵConsole,
+  ɵresetCompiledComponents,
+} from '@angular/core';
+import {
+  ɵSERVER_CONTEXT as SERVER_CONTEXT,
+  renderApplication,
+  renderModule,
+} from '@angular/platform-server';
+import type { AngularServerApp } from './app';
+import { Console } from './console';
+import { REQUEST, REQUEST_CONTEXT, RESPONSE_INIT } from './tokens';
+
+/**
+ * Renders an Angular server application to produce a response for the given HTTP request.
+ * Supports server-side rendering (SSR), static site generation (SSG), or app shell rendering.
+ *
+ * @param app - The server application instance to render.
+ * @param request - The incoming HTTP request object.
+ * @param serverContext - Context specifying the rendering mode.
+ * @param requestContext - Optional additional context for the request, such as metadata.
+ * @returns A promise that resolves to a response object representing the rendered content.
+ */
+export async function render(
+  app: AngularServerApp,
+  request: Request,
+  serverContext: 'ssr' | 'ssg' | 'app-shell',
```

Comment:
I think it might be helpful to have an enum for those values (to avoid typos, etc).